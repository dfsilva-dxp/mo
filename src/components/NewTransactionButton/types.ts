import { AnchorHTMLAttributes, ReactNode } from "react";

export type NewTransactionButtonProps = {
  label: string;
  description?: string;
  icon: ReactNode;
} & AnchorHTMLAttributes<HTMLAnchorElement>;
