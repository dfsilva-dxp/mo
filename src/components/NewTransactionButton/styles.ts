import styled, { css } from "styled-components";

export const Button = styled.a`
  ${({ theme }) => css`
    display: inline-grid;
    grid-template-columns: auto 1fr;
    place-items: center;
    gap: ${theme.spacings.xsmall};
    background: ${theme.colors.panel};
    width: 100%;
    height: 10rem;
    padding: ${theme.spacings.small} 5.6rem;
    border-radius: ${theme.border.radius};
    font-size: ${theme.font.sizes.large};
    cursor: pointer;
    transition: filter ${theme.transition.default};

    &:hover {
      filter: brightness(0.9);
    }
  `}
`;

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-rows: repeat(2, auto);
    place-items: center;
    text-align: center;
    font-size: ${theme.font.sizes.small};

    span {
      font-size: ${theme.font.sizes.xsmall};
      color: ${theme.colors.gray400};
    }
  `}
`;
