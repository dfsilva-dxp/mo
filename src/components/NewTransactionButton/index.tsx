import { NewTransactionButtonProps } from "./types";

import { Button, Wrapper } from "./styles";

export default function NewTransactionButton({
  label,
  description,
  icon,
  ...props
}: NewTransactionButtonProps) {
  return (
    <Button {...props}>
      {icon}
      <Wrapper>
        {label}
        <span>{description}</span>
      </Wrapper>
    </Button>
  );
}
