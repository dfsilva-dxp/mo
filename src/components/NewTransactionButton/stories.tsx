import { ComponentStory, ComponentMeta } from "@storybook/react";
import { TiPlus } from "react-icons/ti";

import NewTransactionButton from ".";

export default {
  title: "NewTransactionButton",
  component: NewTransactionButton,
  argTypes: {},
  parameters: {
    layout: "centered"
  },
  args: {
    label: "Gerar nova transação",
    description: "Movimentações de entradas e saídas",
    icon: <TiPlus />
  },
  decorators: [
    (Story) => (
      <div style={{ width: "37rem" }}>
        <Story />
      </div>
    )
  ]
} as ComponentMeta<typeof NewTransactionButton>;

export const Basic: ComponentStory<typeof NewTransactionButton> = ({
  ...props
}) => <NewTransactionButton {...props} />;
