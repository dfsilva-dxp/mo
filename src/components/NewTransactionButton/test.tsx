import { screen } from "@testing-library/react";
import { TiPlus } from "react-icons/ti";

import { renderWithTheme } from "utils/tests/helpers";

import NewTransactionButton from ".";

describe("ButtonNewTransaction Component", () => {
  it("should render the Button", () => {
    const { container } = renderWithTheme(
      <NewTransactionButton
        label="Gerar nova transação"
        icon={<TiPlus data-testid="icon" />}
        href="/"
      />
    );

    expect(
      screen.getByRole("link", { name: /gerar nova transação/i })
    ).toHaveAttribute("href", "/");

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render a description when is passed in", () => {
    renderWithTheme(
      <NewTransactionButton
        label="Gerar nova transação"
        description="Movimentações de entradas e saídas"
        icon={<TiPlus data-testid="icon" />}
      />
    );

    expect(
      screen.getByText(/Movimentações de entradas e saídas/i)
    ).toBeInTheDocument();
  });

  it("should render an icon when is passed", () => {
    renderWithTheme(
      <NewTransactionButton
        label="Gerar nova transação"
        description="Movimentações de entradas e saídas"
        icon={<TiPlus data-testid="icon" />}
      />
    );

    expect(screen.getByTestId("icon")).toBeInTheDocument();
  });
});
