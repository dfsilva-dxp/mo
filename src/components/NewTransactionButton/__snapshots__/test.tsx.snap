// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonNewTransaction Component should render the Button 1`] = `
.c0 {
  display: inline-grid;
  grid-template-columns: auto 1fr;
  place-items: center;
  gap: 1.6rem;
  background: #1F2029;
  width: 100%;
  height: 10rem;
  padding: 2rem 5.6rem;
  border-radius: 0.4rem;
  font-size: 2.4rem;
  cursor: pointer;
  -webkit-transition: filter 0.15s ease-in-out;
  transition: filter 0.15s ease-in-out;
}

.c0:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c1 {
  display: grid;
  grid-template-rows: repeat(2,auto);
  place-items: center;
  text-align: center;
  font-size: 1.4rem;
}

.c1 span {
  font-size: 1.2rem;
  color: #797D9A;
}

<a
  class="c0"
  href="/"
>
  <svg
    baseProfile="tiny"
    data-testid="icon"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    version="1.2"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18 10h-4v-4c0-1.104-.896-2-2-2s-2 .896-2 2l.071 4h-4.071c-1.104 0-2 .896-2 2s.896 2 2 2l4.071-.071-.071 4.071c0 1.104.896 2 2 2s2-.896 2-2v-4.071l4 .071c1.104 0 2-.896 2-2s-.896-2-2-2z"
    />
  </svg>
  <div
    class="c1"
  >
    Gerar nova transação
    <span />
  </div>
</a>
`;
